### PR TITLE
Classes for patterns states #555

### PR DESCRIPTION
**Pull request by VargBurz**

### CODE REVIEW

File: analytics/analytics/detectors/pattern_detector.py [hunk 1]
```diff
@@ -123,4 +123,4 @@ def consume_data(self, data: pd.DataFrame, cache: Optional[ModelCache]) -> Optio
 
     def get_window_size(self, cache: Optional[ModelCache]) -> int:
         if cache is None: return self.DEFAULT_WINDOW_SIZE
-        return cache.get('WINDOW_SIZE', self.DEFAULT_WINDOW_SIZE)
+        return getattr(cache, 'WINDOW_SIZE', self.DEFAULT_WINDOW_SIZE)
```
Comment (jonyrock) [inline analytics/analytics/detectors/pattern_detector.py]: nonononononononoon 
```
if cache.WINDOW_SIZE is None:
    return self.DEFAULT_WINDOW_SIZE
return cache.WINDOW_SIZE
```

File: analytics/analytics/models/model.py [hunk 2]
```diff
@@ -67,7 +79,7 @@ def get_model_type(self) -> (str, bool):
     def fit(self, dataframe: pd.DataFrame, segments: list, id: str, cache: Optional[ModelCache]) -> ModelCache:
         logging.debug('Start method fit for analytic unit {}'.format(id))
         data = dataframe['value']
-        if cache != None and len(cache) > 0:
+        if cache != None:
```
Comment (jonyrock) [inline analytics/analytics/models/model.py]:
```suggestion
        if cache is not None:
```

File: analytics/analytics/models/model.py [hunk 3]
```diff
@@ -96,7 +108,7 @@ def fit(self, dataframe: pd.DataFrame, segments: list, id: str, cache: Optional[
 
     def detect(self, dataframe: pd.DataFrame, id: str, cache: Optional[ModelCache]) -> dict:
         #If cache is None or empty dict - default parameters will be used instead
-        if cache != None and len(cache) > 0:
+        if cache != None:
```
Comment (jonyrock) [inline analytics/analytics/models/model.py]:
```suggestion
        if cache is not None:
```

File: analytics/analytics/models/model.py [hunk 4]
```diff
@@ -119,11 +131,11 @@ def detect(self, dataframe: pd.DataFrame, id: str, cache: Optional[ModelCache])
         }
 
     def _update_fiting_result(self, state: dict, confidences: list, convolve_list: list, del_conv_list: list, height_list: list) -> None:
-        if type(state) is dict:
-            state['confidence'] = float(min(confidences, default = 1.5))
-            state['convolve_min'], state['convolve_max'] = utils.get_min_max(convolve_list, state['WINDOW_SIZE'])
-            state['conv_del_min'], state['conv_del_max'] = utils.get_min_max(del_conv_list, 0)
-            state['height_min'], state['height_max'] = utils.get_min_max(height_list, 0)
+        if state != None:
```
Comment (jonyrock) [inline analytics/analytics/models/model.py]:
```suggestion
        if state is not None:
```

File: analytics/analytics/models/model.py [hunk 5]
```diff
@@ -43,15 +43,34 @@ def convert_nan_to_zero(self):
...
-    def __init__(self):
-        self.pattern_center = []
-        self.pattern_model = []
-        self.confidence = 0
-        self.convolve_max = 0
-        self.convolve_min = 0
-        self.WINDOW_SIZE = 0
-        self.conv_del_min = 0
-        self.conv_del_max = 0
+    def __init__(self, cache = None):
```
Comment (jonyrock) [inline analytics/analytics/models/model.py]: looks like better type for this is `cache: Optional[dict]`

File: analytics/analytics/models/model.py [hunk 6]
```diff
@@ -41,6 +41,44 @@ def convert_nan_to_zero(self):
...
+        self.convolve_min = cache.get('convolve_min', 0)
+        self.WINDOW_SIZE = cache.get('WINDOW_SIZE', 0)
+        self.conv_del_min = cache.get('conv_del_min', 0)
+        self.conv_del_max = cache.get('conv_del_max', 0)
+        self.JUMP_HEIGHT = cache.get('JUMP_HEIGHT', 0)
+        self.JUMP_LENGTH = cache.get('JUMP_LENGTH', 0)
+        self.height_max = cache.get('height_max', 0)
+        self.height_min = cache.get('height_min', 0)
+        self.DROP_HEIGHT = cache.get('DROP_HEIGHT', 0)
+        self.DROP_LENGTH = cache.get('DROP_LENGTH', 0)
```
Comment (rozetko) [inline analytics/analytics/models/model.py]: Let's make a parent `ModelState` class with general fields and separate children: `DropModelState` / `PeakModelState` / etc
It's strange that we have `DROP_LENGTH` in Peak model, for example

**Comment by VargBurz** [comment]
Splitted on subtasks

**Pull request closed without merging**